packages/@lwc/template-compiler/src/transform/binding.ts

```diff
@@ -212,7 +219,18 @@ function transformToASTNode(element: IRElement): BindingASTNode {
     throw new Error('Element must have either a `for:each`, `iterator:*`, or `if` directive.');
 }
 
-function transformToASTComponentNode(element: IRElement): BindingASTComponentNode {
+function transformToSlotNode(element: IRElement): BindingASTSlotNode {
+    if (isSlot(element)) {
+        return {
+            type: 'BindingASTSlotNode',
+            name: element.slotName || '',
+            children: [],
+        };
+    }
+    throw new Error(`Expected element ${element.tag} to be a slot.`);
```

Comment:
should we say `Expected a <slot> element, instead received ${element.tag}`

Reply:
Updated to `Expected <slot> element but received <${element.tag}> element.`